inter/block_test.go

```diff
@@ -0,0 +1,55 @@
+package inter
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"github.com/Fantom-foundation/lachesis-base/hash"
+	"github.com/ethereum/go-ethereum/common"
+	"math/rand"
+	"testing"
+)
+
+func TestComputePrevRandao_ComputationIsDeterministic(t *testing.T) {
+	events := hash.FakeEvents(5)
+	randao1 := ComputePrevRandao(events)
+	rand.Shuffle(len(events), func(i, j int) {
+		events[i], events[j] = events[j], events[i]
+	})
+	randao2 := ComputePrevRandao(events)
+	if randao1 != randao2 {
+		t.Error("computation is not deterministic")
+	}
+}
+
+func TestComputePrevRandao_ComputationWorks(t *testing.T) {
+	tests := []struct {
```

Comment:
In such a case, where we really want to check the correctness of the hashing function end-to-end I might opt to hard-code inputs and outputs. These results are never expected to change, so the test never needs to be updated (maintenance costs = 0).

As it is right now, it is a 1:1 copy of the implementation code and errors could be easily correlated between the implementation and the test case. If it would be a different implementation, it would be better, but like this this test is to easily missing systematic issues.

For computing reference values I suggest to hard-code random input events, then do the computation of the resulting hashes by hand and use some web-tool for computing the sha256 that is to be expected.